### PR TITLE
Add student service

### LINKED ISSUE #14: create student and prog dept

### PR DESCRIPTION
**Pull request by somyagupta18**
- Closes #14

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -20,3 +20,6 @@ DATABASE:
 AUTH:
   PORT: "8010"
   DBNAME: "auth"
+STUDENT:
+  PORT: "8015"
+  DBNAME: "student"
```
Comment (1-Harshit) [inline config.yaml:25]: Empty line pls

File: student/db.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+func updateStudent(ctx *gin.Context, student *Student, id uint) error {
+	tx := db.WithContext(ctx).Model(&Student{}).Where("id = ?", id).Updates(student)
+	return tx.Error
+}
+
+func getAllStudents(ctx *gin.Context, students *[]Student) error {
+	tx := db.WithContext(ctx).Find(students)
+	return tx.Error
+}
+
```
Comment (1-Harshit) [inline student/db.go:24]: extra line? where is go fmt bruh

File: student/program_dept_var.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+type DepartmentID uint
+
+type ProgramDepartment struct {
+	ID   uint
+	Prog ProgramID
+	Dept DepartmentID
+}
+
+const (
+	BT    ProgramID = 0
```
Comment (1-Harshit) [inline student/program_dept_var.go:14]: - will be fixed in #16

File: student/update.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	}
+
+	err = updateStudent(ctx, &updateStudentRequest, uint(id))
+
+	if err != nil {
+		ctx.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	logrus.Infof("A student %s is updated", updateStudentRequest.IITKEmail)
```
Comment (1-Harshit) [inline student/update.go:32]: Use ID, what if IITK email not present

**Review by 1-Harshit — CHANGES_REQUESTED** [review]

**Review by 1-Harshit — APPROVED** [review]
LGTM

**Comment by abhishekshree** [comment]
LGTM

**Pull request merged**